grails-app/services/com/unifina/service/ProductService.groovy

```diff
@@ -184,18 +184,18 @@ class ProductService {
 
 	void addStreamToProduct(Product product, Stream stream, SecUser currentUser)
 			throws ValidationException, NotPermittedException {
-		permissionService.verifyShare(currentUser, stream)
+		permissionService.verifyShareStream(currentUser, stream)
 		product.streams.add(stream)
 		product.save(failOnError: true)
 		// A stream that is added when editing an existing free product should inherit read access for anonymous user
 		if (product.isFree()) {
-			permissionService.systemGrantAnonymousAccess(stream, Permission.Operation.READ)
+			permissionService.systemGrantAnonymousAccess(stream, Permission.Operation.STREAM_GET)
 		}
 		if (product.type == Product.Type.COMMUNITY) {
 			Set<SecUser> users = communityJoinRequestService.findCommunityMembers(product.beneficiaryAddress)
 			for (SecUser u : users) {
-				if (!permissionService.canWrite(u, stream)) {
-					permissionService.systemGrant(u, stream, Permission.Operation.WRITE)
+				if (!permissionService.canWriteStream(u, stream)) {
```

Comment:
We should check `STREAM_PUBLISH` here.

As we know it's a stream, just doing something like 

`permissionService.check(u, stream, Operation.STREAM_PUBLISH)` 

avoids the need type-specific method boilerplate in `PermissionService`.